data_structures/src/chain.rs

```diff
@@ -1701,6 +1722,24 @@ impl TransactionsPool {
     /// assert!(!pool.dr_contains(&transaction.hash()));
     /// ```
     pub fn dr_remove(&mut self, key: &Hash) -> Option<DRTransaction> {
```

Comment:
/// Remove a data request transaction from the pool and make sure that other transactions that may try to spend the same UTXOs are also removed.